phoenix-core/src/main/java/org/apache/phoenix/cache/GlobalCache.java

```diff
@@ -157,4 +159,18 @@ public TenantCache getChildTenantCache(ImmutableBytesWritable tenantId) {
         }
         return tenantCache;
     }
+
+    public static class FunctionBytesPtr extends ImmutableBytesPtr {
+
+        public FunctionBytesPtr(byte[] key) {
+            super(key);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
I will override the hashcode and just call super.hashCode() then we won't get checkstyle or findbug warning.